luxonis_train/nodes/heads/ddrnet_segmentation_head.py

```diff
@@ -0,0 +1,94 @@
+"""DDRNet segmentation head.
+
+Adapted from: U{https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/models/segmentation_models/ddrnet.py}
+Original source: U{https://github.com/ydhongHIT/DDRNet}
+Paper: U{https://arxiv.org/pdf/2101.06085.pdf}
+@license: U{https://github.com/Deci-AI/super-gradients/blob/master/LICENSE.md}
+"""
+
+import torch.nn as nn
+from torch import Tensor
+
+from luxonis_train.nodes.base_node import BaseNode
+from luxonis_train.nodes.blocks import ConvModule
+from luxonis_train.utils.types import LabelType
+
+
+class DDRNetSegmentationHead(BaseNode[Tensor, Tensor]):
+    in_height: int
+    in_channels: int
+    tasks: list[LabelType] = [LabelType.SEGMENTATION]
+
+    def __init__(
+        self,
+        num_classes: int,
```

Comment:
Similarly to above, there are already defined properties `n_classes` and `in_channels` that can be used instead of `num_classes` and `in_planes` (and I suppose the default value of `inter_planes` should be then computed based on `in_channels`).

Reply:
I would keep `in_planes`, `inter_planes` and `scale_factor` as separate arguments, as in the original implementation. Not sure how to use `in_channels` instead of `in_planes`.

Reply:
`BaseNode.in_channels` is exactly the same thing as what you're using the `in_planes` for, that is the number of channels of the input to the head. Only there is the additional value of being always set to the right number (in order for the shapes in the convolution layers to match) instead of relying on the user to provide it (who in our case is neither a programmer nor knowledgeable in ML). 

So, in short you just replace `in_planes` with `self.in_channels` and it will work just the same but won't break if the shape of the input changes. And you can also remove the `in_planes` computation from the predefined model. 

Specific example for why to do it this way; if I were to use this head with the resnet backbone, it will fail because the backbone produces outputs with 2048 channels, but the head expects 128. So I as a user would have to figure out what to change and how in order for the model to work. However, by using the `BaseNode` properties, I can attach this head to any backbone and it simply :magic_wand: _works_ :magic_wand: 

As a rule of thumb; any purely technical parameter that must be set correctly in order for the shapes to match (like `in_planes` or `scale_factor`) should not be required from the user.